src/subcommand.rs

```diff
@@ -485,12 +487,37 @@ impl Subcommand {
         let mut line_width = UnicodeWidthStr::width(*name);
 
         for parameter in &recipe.parameters {
-          if let Some(evaluator) = &mut evaluator {
-            let parameter = parameter.evaluate_default(evaluator)?;
-          };
-          line_width += UnicodeWidthStr::width(
-            format!(" {}", parameter.color_display(Color::never())).as_str(),
-          );
+          // FIXME: unwrap -> ?
+          if let Some(evaluated) = evaluator
+            .as_mut()
+            .map(|evaluator| parameter.evaluate_default(evaluator))
+            .transpose()
+            .unwrap()
+            .flatten()
+          {
+            use std::collections::btree_map::Entry;
+
+            line_width += UnicodeWidthStr::width(
+              format!(" {}", evaluated.color_display(Color::never())).as_str(),
+            );
+
+            match evaluated_parameters.entry(name) {
+              Entry::Vacant(vacant) => {
+                let mut parameters = BTreeMap::new();
+                parameters.insert(evaluated.name.lexeme(), evaluated);
+                vacant.insert(parameters);
+              }
+              Entry::Occupied(mut occopied) => {
+                occopied
+                  .get_mut()
+                  .insert(evaluated.name.lexeme(), evaluated);
+              }
+            }
```

Comment:
I've tested this locally with a justfile like this:
```make
key1 := `echo val11`
key2 := `echo val22`

recipe1 arg1=key1 arg2=`echo val12`:
    echo

recipe2 arg1=`echo val21` arg2=key2:
    echo
```
But I wasn't sure how to exactly write a test here to fit in with the rest, if that is even necessary.